File: 680. Valid Palindrome II/680. Valid Palindrome II.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        for i in range((len(s))//2):
            if s[i] != s[-1-i]:
                t, u = s[:i]+s[i+1:], s[:-1-i]+s[len(s)-i:]
                return t == t[::-1] or u == u[::-1]
        return True

File: 680. Valid Palindrome II/680. Valid Palindrome II.java
class Solution {
    public boolean validPalindrome(String s) {
        int start = 0;
        int end = s.length()-1;
        while(start<=end){
            if(s.substring(start,start+1).equals(s.substring(end,end+1))){
                start += 1;
                end -= 1;
            }else{
                return(isTrue(s,start+1,end)||isTrue(s,start,end-1));
            }
        }
        return true;
    }
    public boolean isTrue(String s, Integer start, Integer end){
        while(start<=end){
            if(s.substring(start,start+1).equals(s.substring(end,end+1))){
                start += 1;
                end -= 1;
            }else{
                return false;
            }
        }
        return true;
    }
}